manual/src/refman/extensions/bindingops.etex

```diff
@@ -148,3 +139,80 @@ end
 
 where "(let*)" is bound to the "bind" operation, and "(and*)" is also
 bound to the monoidal product operation.
+
+\subsection{ss:letop-rules}{General desugaring rules}
+
+The form
+
+\begin{verbatim}
+let<op0>
+  x1 = e1
+and<op1>
+  x2 = e2
+and<op2>
+  x3 = e3
+in e
+\end{verbatim}
+
+desugars into
+
+\begin{verbatim}
+( let<op0> )
+  (( and<op2> )
+    (( and<op1> )
+      e1
+      e2)
+    e3)
+  (fun ((x1, x2), x3) -> e)
+\end{verbatim}
+
+This of course works for any number of nested "and"-operators. One can
+express the general rule by repeating the following simplification
+steps:
+\begin{itemize}
+\item
+The first "and"-operator in
+\begin{center}
+"let<op0> x1 = e1 and<op1> x2=e2 and... in e"
```

Comment:
```suggestion
"let<op0> x1 = e1 and<op1> x2 = e2 and ... in e"
```

Reply:
In my mind `and...` can be read as a metavariable for a list of and<op>-bindings, which is why I omitted the spaces there. (Technically we need this rule to apply even in situations where there are exactly two bindings, so `and...` is empty and does not start with `and`.) I don't know if I should keep this quirk or not. I'll definitely fix `x2=e2`, we can't have that. Thanks!

Reply:
Fair enough, go with what you feel!